service/core.go

```diff
@@ -35,6 +36,7 @@ import (
 
 // A Host represents the hosting environment for a service instance
 type Host interface {
+	AuthClient() uauth.AuthClient
```

Comment:
TBD: is `Auth` too short? Do we mean that we encapsulate both Authorization **and** Authentication? If so, this is probably appropriate. If not, we should be explicit.